From: Michael Catanzaro <mcatanzaro@posteo.net>
Date: Mon, 24 Sep 2018 20:19:26 +0000
Subject: wayland: Avoid crashes inside wl_proxy_marshal()

Suggested by Garnacho. Hopefully fixes #1349.

Note: I'm riskily committing this via web UI not because I'm lazy
(though I am :) but because I'm seeing a weird host key when I try to
push or pull from GitLab.

Origin: upstream, 3.24.2, commit:b4464e12742767a3dc08317371d665854286355f
---
 gdk/wayland/gdkwindow-wayland.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index 5f16277..a5f9d44 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -932,6 +932,9 @@ gdk_window_impl_wayland_end_paint (GdkWindow *window)
   cairo_rectangle_int_t rect;
   int i, n;
 
+  if (!GDK_WINDOW_IS_MAPPED (window))
+    return;
+
   if (impl->staging_cairo_surface &&
       _gdk_wayland_is_shm_surface (impl->staging_cairo_surface) &&
       !window->current_paint.use_gl &&
